extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-module.ts

```diff
@@ -54,6 +54,16 @@ export class DataExtensionsEditorModule {
   public getCommands(): DataExtensionsEditorCommands {
     return {
       "codeQL.openDataExtensionsEditor": async () => {
+        if (
+          !(await this.cliServer.cliConstraints.supportsResolveExtensions())
+        ) {
+          void showAndLogErrorMessage(
+            this.app.logger,
+            "CodeQL CLI version v2.10.2 or later is required to use the data extensions editor.",
+          );
+          return;
+        }
+
```

Comment:
Could we move this check to be in the same location as the other check (`supportsQlpacksKind`)? Could we then also use `CLI_VERSION_WITH_RESOLVE_EXTENSIONS.format()` instead of hardcoding `v2.10.2`?

Reply:
Yep, sorry I didn't spot that we're already doing a CLI version check here.

We now have two checks next to each other. I've put the one with the stricter version requirement first, it removes the possibility of upgrading to pass the first check and then just hitting the second check.

So actually we were already protected because we already require `v2.12.3`. I'm tempted to leave both checks in, even though the second one is incapable of failing, because it acts as internal documentation that we're using a particular feature. When the CodeQL CLI version advances enough that we can assume `v.2.10.2` then we can remove the check for that.

Do you think this makes sense or does it offer no benefit and we should just remove the weaker version check?

Reply:
That makes sense to me, it makes sense to be explicit about which features (and thus versions) we require.